### PR TITLE
[qa] Add test for rescan feature of wallet key import RPCs

### PR DESCRIPTION
**Pull request by ryanofsky**
Test only change.

New test covers importaddress, importpubkey, importprivkey, and importmulti RPC's.

This was originally part of #9137, but I split it off because the test is useful on its own.

### CODE REVIEW

File: qa/rpc-tests/import-rescan.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2016 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
```
Comment (MarcoFalke) [inline qa/rpc-tests/import-rescan.py]: nit: no wildcard import here.

**Review by maflcko — COMMENTED** [review]
Concept ACK

File: qa/rpc-tests/import-rescan.py (hunk 1 above)
Comment (ryanofsky) [inline qa/rpc-tests/import-rescan.py]: Removed

**Review by maflcko — APPROVED** [review]
utACK 7e1d6fd43659e7cbfce2efd5433d79ef0fe687ec

(Sorry for sending nits your way a second time, feel free to not fix them; They are mostly not mandatory for the test to function.)

File: qa/rpc-tests/import-rescan.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2016 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (start_nodes, connect_nodes, sync_blocks,
+                                 assert_equal, Decimal)
```
Comment (ryanofsky) [inline qa/rpc-tests/import-rescan.py]: Fixed.

File: qa/rpc-tests/import-rescan.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+        # Create spends for all the imported addresses.
+        spend_txids = []
+        for import_rpc, label, amount, txid, addr in import_rpc_variants:
+            raw_tx = self.nodes[0].getrawtransaction(txid)
+            decoded_tx = self.nodes[0].decoderawtransaction(raw_tx)
+            input_vout = next(
+                out["n"] for out in decoded_tx["vout"]
+                if out["scriptPubKey"]["addresses"] == [addr["address"]])
+            fee = self.nodes[0].getnetworkinfo()["relayfee"]
```
Comment (ryanofsky) [inline qa/rpc-tests/import-rescan.py]: Done.

File: qa/rpc-tests/import-rescan.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+            inputs = [{"txid": txid, "vout": input_vout}]
+            outputs = {self.nodes[0].getnewaddress(): Decimal(amount) - fee}
+            raw_spend_tx = self.nodes[0].createrawtransaction(inputs, outputs)
+            signed_spend_tx = self.nodes[0].signrawtransaction(raw_spend_tx)
+            spend_txid = self.nodes[0].sendrawtransaction(signed_spend_tx[
+                "hex"])
+            spend_txids.append(spend_txid)
+
+        self.nodes[0].generate(1)
+        sync_blocks(self.nodes)
```
Comment (ryanofsky) [inline qa/rpc-tests/import-rescan.py]: Done.

**Comment by laanwj** [comment]
Thanks, LGTM 
utACK d8c0b9f

**Pull request merged**